src/RustyObjectStore.jl

```
@@ -555,4 +559,53 @@ function put_object(buffer::AbstractVector{UInt8}, path::String, conf::AbstractC
     end
 end
 
+"""
+    delete_object(path, conf)
+
+Send a delete request to the object store.
+
+# Arguments
+- `path::String`: The location of the object to delete.
+- `conf::AbstractConfig`: The configuration to use for the request.
+  It includes credentials and other client options.
+
+# Throws
+- `DeleteException`: If the request fails for any reason. Note that S3 will treat a delete request
+  to a non-existing object as a success, while Azure Blob will treat it as a 404 error.
+"""
+function delete_object(path::String, conf::AbstractConfig)
+    response_ref = Ref(Response())
+    cond = Base.AsyncCondition()
+    cond_handle = cond.handle
+    config = into_config(conf)
+    while true
+        result = @ccall rust_lib.delete(
+            path::Cstring,
+            config::Ref{Config},
+            response_ref::Ref{Response},
+            cond_handle::Ptr{Cvoid}
+        )::Cint
+
+        if result == 1
+            @error "failed to submit put, internal channel closed"
```

throw instead of log